cmd/goplugin-heroku/resource/build.go

```diff
@@ -253,3 +266,25 @@ func uploadSource(filePath, httpMethod, httpUrl string) error {
 
 	return nil
 }
+
+func buildpackCheck(desiredState Build, currentState heroku.Build) bool {
+	var status []bool
+	var result bool
+
+	for i, packs := range desiredState.Buildpacks {
+		if strings.Contains(currentState.Buildpacks[i].URL, *packs.Name) {
```

Comment:
Is `len(desiredState.Buildpacks) <= len(currentState.Buildpacks)` guaranteed? If not, then this might fail with an out of bounds panic. Also, if `len(desiredState.Buildpacks) < len(currentState.Buildpacks)` can be true, then I guess the excess entries in `currentState` should affect the outcome somehow.

Reply:
Good point on the first part. There is a chance that `len(desiredState.Buildpacks) > len(currentState.Buildpacks)`, at which point we'd want an automatic false generated to indicate that the desired state has more buildpacks than exist on the app requiring an update.

As for the second part, the tricky part with buildpacks is that an app always requires a buildpack, but the user doesn't have to explicitly name one. If one isn't named, heroku will attempt to figure out what is needed based on what the build contains. So `currentState` will always have a buildpack if a build exists, but the `desiredState` doesn't have to have one defined.

That said, it'd probably be a good idea to check `len(desiredState.Buildpacks)` and if it is `= 0`, then automatically ignore the Buildpacks field. The other option would be to require that a user define a buildpack to make lyra's life easier

Reply:
When `len(desiredState.Buildpacks) > len(currentState.Buildpacks)`, this line will still panic.

Reply:
It shouldn't ever reach that line if `len(desiredState.Buildpacks > len(currentState.Buildpacks)`. 

See https://github.com/lyraproj/lyra/pull/342/files#diff-de40d7ec6f34bc2a1d332f56d2c58ddaR328